### PR TITLE
Update net-mail/b4 to 0.10.1 and convert dev-vcs/git-filter-repo to distutils-r1

### PR DESCRIPTION
**Pull request by palmer-dabbelt**
This updates b4 to the latest release.  b4 now depends on dev-vcs/git-filter-repo as a python module, so move that to distutils-r1 rather than trying to retrofit modules into the old non-standard package flow (which I couldn't get working).

### CODE REVIEW

**Comment by palmer-dabbelt** [comment]
I'm not really sure what I'm doing here, but I think the CI failure is incorrect: it says `nonsolvable depset(rdepend) keyword(~amd64) stable profile (default/linux/amd64/17.1) (24 total): solutions: [ dev-vcs/git-filter-repo[python_targets_python3_10(-),python_targets_python3_8(-),python_targets_python3_9(-)] ]`, but that should be resolved by the previous changes to `git-filter-repo` -- at least it is locally and that's what I was trying to do here.

Not sure if I'm misunderstanding something, though.

File: net-mail/b4/b4-0.10.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Copyright 2020-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_SETUPTOOLS=rdepend
+PYTHON_COMPAT=( python3_{8,9,10} )
```
Comment (thesamesam) [inline net-mail/b4/b4-0.10.1.ebuild:7]: I think the issue is that we've kept Python 3.8 here but no versions of git-filter-repo (a dependency of b4) have Python 3.8 support.

File: net-mail/b4/b4-0.10.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Copyright 2020-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_SETUPTOOLS=rdepend
```
Comment (thesamesam) [inline net-mail/b4/b4-0.10.1.ebuild:6]: You can drop this now -- maybe try PEP517: https://projects.gentoo.org/python/guide/distutils.html#the-pep-517-and-legacy-modes.

It's a nuisance but it's been foisted upon us by the Python ecosystem.

File: dev-vcs/git-filter-repo/git-filter-repo-2.38.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+DESCRIPTION="Quickly rewrite git repository history (filter-branch replacement)"
+HOMEPAGE="https://github.com/newren/git-filter-repo/"
+SRC_URI="
+	https://github.com/newren/git-filter-repo/releases/download/v${PV}/${P}.tar.xz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+REQUIRED_USE=${PYTHON_REQUIRED_USE}
```
Comment (thesamesam) [inline dev-vcs/git-filter-repo/git-filter-repo-2.38.0.ebuild:17]:
```suggestion
REQUIRED_USE="${PYTHON_REQUIRED_USE}"
```

File: dev-vcs/git-filter-repo/git-filter-repo-2.34.0.ebuild [hunk 4]
```diff
@@ -4,7 +4,7 @@
 EAPI=8
 
 PYTHON_COMPAT=( python3_{9..10} )
```
Comment (thesamesam) [inline dev-vcs/git-filter-repo/git-filter-repo-2.34.0.ebuild:6]: Given this affects installed files, I'd revbump the older versions (just git mv if you want), or just only bother touching 2.38.0.

File: net-mail/b4/b4-0.10.1.ebuild (hunk 2 above)
Comment (palmer-dabbelt) [reply net-mail/b4/b4-0.10.1.ebuild]: I have no idea what that means, but it still builds with `DISTUTILS_USE_PEP517=setuptools`.  I've squashed that one in.

File: net-mail/b4/b4-0.10.1.ebuild (hunk 1 above)
Comment (palmer-dabbelt) [reply net-mail/b4/b4-0.10.1.ebuild]: WFM, I've squashed in a fix to drop 3.8.

File: dev-vcs/git-filter-repo/git-filter-repo-2.38.0.ebuild (hunk 3 above)
Comment (palmer-dabbelt) [reply dev-vcs/git-filter-repo/git-filter-repo-2.38.0.ebuild]: Thanks, this should be fixed.

File: dev-vcs/git-filter-repo/git-filter-repo-2.34.0.ebuild (hunk 4 above)
Comment (palmer-dabbelt) [reply dev-vcs/git-filter-repo/git-filter-repo-2.34.0.ebuild:6]: Not 100% sure what you're saying, but I squashed in a rename of these to `2.3{3,4}.0-r1`.  IIUC that'll trigger a rebuild for users.

**Comment by palmer-dabbelt** [comment]
I also noticed that 2.38.0 landed, I rebased on master and squashed in the -r1.  It's 2 commits now, as it's just the distutils conversion for all 3.

**Comment by thesamesam** [comment]
All done, thanks!

**Pull request closed without merging**